src/codegeneration/JsxTransformer.js

```diff
@@ -80,18 +82,52 @@ export class JsxTransformer extends ParseTreeTransformer {
 
   transformJsxElement(tree) {
     let name = this.transformAny(tree.name);
-    let attrs = this.transformList(tree.attributes);
+    let props = this.transformJsxAttributes_(tree);
     let children = this.transformJsxChildren_(tree.children);
-    let props;
-    if (attrs.length > 0) {
-      props = createObjectLiteral(attrs);
-    } else {
-      props = createNullLiteral();
-    }
     let args = createArgumentList([name, props, ...children]);
     return parseExpression `${this.getJsxFunction_()}(${args})`;
   }
 
+  transformJsxAttributes_(tree) {
+    let attrs = this.transformList(tree.attributes);
+    if (attrs.length === 0) {
+      return createNullLiteral();
+    }
+    if (tree.attributes.some(a => a.type === JSX_SPREAD_ATTRIBUTE)) {
+      return this.createSpreadAttributeExpression_(attrs);
+    }
+    return createObjectLiteral(attrs);
+  }
+
+  createSpreadAttributeExpression_(attrs) {
+    // <a b='b' c='cc {...d} {...g} />
+    // =>
+    // {b: 'b', c: 'c'}, d, g
+
+    let args = [];
+    let currentObject = null;
+    for (let i = 0; i < attrs.length; i++) {
+      let attr = attrs[i];
+      if (attr.type === PROPERTY_NAME_ASSIGNMENT) {
```

Comment:
I think a comment would be helpful here.  We could potentially take different branches for different attributes
and we are updating `args` and `currentObject`.  Or maybe rename `currentObject` to `accumulatePropertyNames` or something.